jung-algorithms/src/main/java/edu/uci/ics/jung/layout/algorithms/TreeLayoutAlgorithm.java

```diff
@@ -74,11 +74,11 @@ public TreeLayoutAlgorithm(int distx, int disty) {
     Preconditions.checkArgument(disty >= 1, "Y distance must be positive");
     this.distX = distx;
     this.distY = disty;
-    this.m_currentPoint = pointModel.newPoint(0, 0);
   }
 
   public void visit(LayoutModel<N, P> layoutModel) {
     super.visit(layoutModel);
+    this.m_currentPoint = pointModel.newPoint(0, 0);
```

Comment:
Unless I'm missing something, this assignment never has any effect, because it's immediately overwritten inside the buildTree() call, so we can just remove it.

Pushing this even farther: it's not at all clear to me that we even want a currentPoint instance field.  Seems like it would be better to make the following changes:

(1) change the buildTree overload with three arguments so that the last argument is the current point (not just its X coordinate)
(2) add a P argument to setCurrentPositionFor() and pass the current point to it

That way we don't need this piece of class-level state at all, which seems like a better/more robust design.

Reply:
Ah yes, TreeLayout. It was one of our contributed Layouts and has always been a little odd. There's probably a ticket out there somewhere to completely redesign it.